typescript/sdk/src/deploy/verify/ContractVerifier.ts

```diff
@@ -338,29 +339,35 @@ export class ContractVerifier {
   async getContractVerificationStatus(
     chain: ChainName,
     address: Address,
-    verificationLogger: Logger,
-  ): Promise<{ isVerified: false } | { isVerified: true; name: string }> {
-    verificationLogger.trace(
-      `Fetching contract ABI for ${chain} address ${address}`,
-    );
-    const sourceCodeResults = (
-      await this.submitForm(
-        chain,
-        ExplorerApiActions.GETSOURCECODE,
-        verificationLogger,
-        { address },
-      )
-    )[0]; // This specific query only returns 1 result
-
-    // Explorer won't return ContractName if unverified
-    return sourceCodeResults.ContractName
-      ? {
-          isVerified: true,
-          name: sourceCodeResults.ContractName,
-        }
-      : {
-          isVerified: false,
-        };
+    verificationLogger: Logger = this.logger,
+  ): Promise<ContractVerificationStatus> {
+    try {
+      const metadata = this.multiProvider.tryGetChainMetadata(chain);
+      const rpcUrl = metadata?.rpcUrls[0].http ?? '';
+      if (rpcUrl.includes('localhost') || rpcUrl.includes('127.0.0.1')) {
+        verificationLogger.debug('Skipping verification for local endpoints');
+        return ContractVerificationStatus.Skipped;
+      }
+      verificationLogger.trace(
+        `Fetching contract ABI for ${chain} address ${address}`,
+      );
+      const sourceCodeResults = (
+        await this.submitForm(
+          chain,
+          ExplorerApiActions.GETSOURCECODE,
+          verificationLogger,
+          { address },
+        )
+      )[0]; // This specific query only returns 1 result
+
+      // Explorer won't return ContractName if unverified
+      return sourceCodeResults.ContractName
+        ? ContractVerificationStatus.Verified
+        : ContractVerificationStatus.Unverified;
+    } catch (e) {
+      this.logger.info(`Error fetching contract verification status: ${e}`);
```

Comment:
Can you add also the chain nameand the contract address in the log? it might be useful for debugging